_health-care/_js/_components/_health-care-app.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import { Nav } from './_hello-world.jsx';
+
+class HealthCareApp extends React.Component {
+  constructor() {
+    super();
+    this.onStateChange = this.onStateChange.bind(this);
+    this.state = { applicationData: {} };
+    const savedAppData = window.localStorage["health-app-data"];
+    if (savedAppData) {
+      this.state.applicationData = JSON.parse(savedAppData);
+    }
+    if (this.state.applicationData.sharedText === undefined) {
+      this.state.applicationData.sharedText = "shared";
+    }
+    if (this.state.applicationData.happyDate === undefined) {
+      this.state.applicationData.happyDate = {
+        month: "1",
+        day: "15",
+        year: "1997",
+      };
+    }
+    if (this.state.applicationData.sadDate === undefined) {
+      this.state.applicationData.sadDate = {
+        month: "4",
+        day: "5",
+        year: "2014",
+      };
+    }
+  }
+
+  onStateChange(field, update) {
+    let newApplicationData = Object.assign({}, this.state.applicationData);
+    newApplicationData[field] = update;
+    window.localStorage["health-app-data"] = JSON.stringify(newApplicationData);
```

Comment:
In react, you're supposed to consider `this.state` to be immutable and only update it via `this.setState()`. Thus, I made a copy.